packages/service/src/utils.ts

```diff
@@ -263,3 +278,274 @@ export function determineNativeContext(
+        }
+    }
+}
+
+/**
+ * Create the storybook capabilities based on the specified browsers
+ */
+export function createStorybookCapabilities(capabilities: Capabilities.RemoteCapabilities, log: Logger) {
+    const isHeadless = process.argv.includes('--headless')
+    const browsers = process.argv.includes('--browsers') ? process.argv[process.argv.indexOf('--browsers') + 1].split(',') : ['chrome']
+
+    if (Array.isArray(capabilities)) {
+        const chromeCapability = {
+            browserName: 'chrome',
+            'goog:chromeOptions': {
+                args: [
+                    'disable-infobars',
+                    ...(isHeadless ? ['--headless'] : []),
+                ],
+            },
+            'wdio-ics:options': {
+                logName: 'local-chrome',
+            },
+        }
+        const firefoxCapability = {
+            browserName: 'firefox',
+            'moz:firefoxOptions': {
+                args: [...(isHeadless ? ['-headless'] : []),]
+            },
+            'wdio-ics:options': {
+                logName: 'local-firefox',
+            },
+        }
+        const safariCapability = {
+            browserName: 'safari',
+            'wdio-ics:options': {
+                logName: 'local-safari',
+            },
+        }
+        const edgeCapability = {
+            browserName: 'MicrosoftEdge',
+            'ms:edgeOptions': {
+                args: [...(isHeadless ? ['--headless'] : [])]
+            },
+            'wdio-ics:options': {
+                logName: 'local-edge',
+            },
+        }
+
+        // Create a map for easy lookup
+        const capabilityMap: CapabilityMap = {
+            chrome: chromeCapability,
+            firefox: firefoxCapability,
+            safari: safariCapability,
+            edge: edgeCapability,
+        }
+
+        // Create new capabilities based on the specified browsers
+        const newCapabilities = browsers
+            .filter((browser): browser is keyof CapabilityMap => browser in capabilityMap)
+            .map(browser => capabilityMap[browser])
+
+        capabilities.length = 0
+        // Add the new capability to the capabilities array
+        // @ts-ignore
+        capabilities.push(...newCapabilities)
+    } else {
+        log.error('The capabilities are not an array')
+    }
+}
+
+/**
+ * Scan the storybook instance
+ */
+export async function scanStorybook(
+    config:Options.Testrunner,
+    log: Logger,
+    options:ClassOptions
+): Promise<{storiesJson: Stories, storybookUrl: string, tempDir: string}>{
+    // Prepare storybook scanning
+    const rawStorybookUrl = process.env.STORYBOOK_URL ?? options?.storybook?.url ?? 'http://127.0.0.1:6006'
+    await checkStorybookIsRunning(rawStorybookUrl)
+    const storybookUrl = sanitizeURL(rawStorybookUrl)
+
+    // Create a temporary folder for test files and add that to the specs
+    const tempDir = temporaryDirectory()
```

Comment:
us Node.js native function for this:
```suggestion
    const tempDir = path.resolve(os.tmpdir(), `wdio-storybook-tests-${Date.now()}`)
```

Do we clean up these tests afterwards?

Reply:
Thanks, didn't knew this, will update this next time

> Do we clean up these tests afterwards?

Yes, in the `onComplete` hook

Reply:
Fixed it